src/software/robot_diagnostics/widgets/slider.cpp

```diff
@@ -0,0 +1,52 @@
+#include "software/robot_diagnostics/widgets/slider.h"
+
+void setupSliderLineEdit(QLineEdit *line_edit, QSlider *slider,
```

Comment:
this is a really useful util, id assume the visualizer has something similar as well, is it possible to share these common UI components between the visualizer and the diagnostics?

Reply:
It is possible and sharing should be encouraged, although the Visualizer doesn't have anything quite like this yet. Once this initial version is merged we will likely want to re-structure our GUI code to make sharing these widgets a bit easier.